src/amp-story-player/amp-story-player-impl.js

```diff
@@ -330,15 +302,30 @@ export class AmpStoryPlayer {
       return;
     }
 
-    this.stories_ = toArray(this.element_.querySelectorAll('a'));
-
+    this.initializeStories_();
     this.initializeShadowRoot_();
     this.initializeIframes_();
     this.initializeButton_();
     this.signalReady_();
     this.isBuilt_ = true;
   }
 
+  /** @private */
+  initializeStories_() {
+    const anchorEls = toArray(this.element_.querySelectorAll('a'));
+
+    anchorEls.forEach((anchorEl) => {
+      const storyObj = /** @type {!StoryDef} */ ({
+        href: anchorEl.href,
+        title: (anchorEl.textContent && anchorEl.textContent.trim()) || null,
+        poster: anchorEl.getAttribute('data-poster-portrait-src'),
+        iframeIdx: -1,
+      });
+
+      this.stories_.push(storyObj);
```

Comment:
Nit: you can use a `map`

Reply:
I meant building the array using Array.prototype.map instead of the forEach yeah